trunk/SUAVE/Methods/Noise/Fidelity_One/Propeller/compute_broadband_noise.py

```diff
@@ -1,123 +1,349 @@
+    delta_r            = np.zeros_like(r)
+    del_r              = r[1:] - r[:-1]
+    delta_r[0]         = 2*del_r[0]
+    delta_r[-1]        = 2*del_r[-1]
+    delta_r[1:-1]      = (del_r[:-1]+ del_r[1:])/2
+
+
+    if np.all(Omega == 0):
+        res.p_pref_broadband                          = np.zeros((num_cpt,num_mic,num_rot,num_cf)) 
+        res.SPL_prop_broadband_spectrum               = np.zeros_like(res.p_pref_broadband)
+        res.SPL_prop_broadband_spectrum_dBA           = np.zeros_like(res.p_pref_broadband)
+        res.SPL_prop_broadband_1_3_spectrum           = np.zeros((num_cpt,num_mic,num_rot,num_cf))
+        res.SPL_prop_broadband_1_3_spectrum_dBA       = np.zeros((num_cpt,num_mic,num_rot,num_cf))
+        res.p_pref_azimuthal_broadband                = np.zeros((num_cpt,num_mic,num_rot,num_azi,num_cf))
+        res.p_pref_azimuthal_broadband_dBA            = np.zeros_like(res.p_pref_azimuthal_broadband)
+        res.SPL_prop_azimuthal_broadband_spectrum     = np.zeros_like(res.p_pref_azimuthal_broadband)
+        res.SPL_prop_azimuthal_broadband_spectrum_dBA = np.zeros_like(res.p_pref_azimuthal_broadband)
+    else:
+        delta        = np.zeros((num_cpt,num_mic,num_rot,num_sec,num_azi,num_cf,2)) #  control points ,  number rotors, number blades , number sections , sides of airfoil
+        delta_star   = np.zeros_like(delta)
+        dp_dx        = np.zeros_like(delta)
+        tau_w        = np.zeros_like(delta)
+        Ue           = np.zeros_like(delta)
+        Theta        = np.zeros_like(delta)
 
+        lower_surface_theta      = np.zeros((num_sec,num_azi))
+        lower_surface_delta      = np.zeros_like(lower_surface_theta)
+        lower_surface_delta_star = np.zeros_like(lower_surface_theta)
+        lower_surface_cf         = np.zeros_like(lower_surface_theta)
+        lower_surface_Ue         = np.zeros_like(lower_surface_theta)
+        lower_surface_H          = np.zeros_like(lower_surface_theta)
+        lower_surface_dp_dx      = np.zeros_like(lower_surface_theta)
+        upper_surface_theta      = np.zeros_like(lower_surface_theta)
+        upper_surface_delta      = np.zeros_like(lower_surface_theta)
+        upper_surface_delta_star = np.zeros_like(lower_surface_theta)
+        upper_surface_cf         = np.zeros_like(lower_surface_theta)
+        upper_surface_Ue         = np.zeros_like(lower_surface_theta)
+        upper_surface_H          = np.zeros_like(lower_surface_theta)
+        upper_surface_dp_dx      = np.zeros_like(lower_surface_theta)
+
+        # ------------------------------------------------------------
+        # ****** TRAILING EDGE BOUNDARY LAYER PROPERTY CALCULATIONS  ******
+        for i in range(num_cpt) : # lower surface is 0, upper surface is 1
+            TE_idx  =  4  # assume trailing edge is the forth from last panel
+
+            if propeller.nonuniform_freestream: # CORRECT THIS HANDLES AZIMUTHALLY CHANGING IN FLOW
```

Comment:
Removed the comment, I coded it but decided not to compute the azimuthal varying noise as yet, I need to handle an issue I am facing with computing speed and memory. This will be future work